content-src/selectors/selectors.js

```diff
@@ -129,7 +135,8 @@ module.exports.selectNewTabSites = createSelector(
 const selectSiteIcon = createSelector(
   site => site,
   site => {
-    const favicon = site.favicon_url || site.favicon;
+    const metadataFavicon = site.favicons && site.favicons[0] && site.favicons[0].url;
+    const favicon = site.favicon_url || metadataFavicon || site.favicon;
```

Comment:
If this

https://github.com/sarracini/activity-stream/blob/21a364cfb68abb389c3cd78e576d32c9f3cde818/content-src/selectors/selectors.js#L31-L35

and this

https://github.com/sarracini/activity-stream/blob/21a364cfb68abb389c3cd78e576d32c9f3cde818/content-src/selectors/selectors.js#L138-L141

share a bunch of logic, can we pull that into a separate function?